js/index.js

```diff
@@ -2,22 +2,31 @@
 
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  let price = 0;
+  let quantity = 0;
+  price = product.querySelector('.price span').innerText;
+  quantity = product.querySelector('.quantity input').value;
+  let subtotal = price * quantity;
+  product.querySelector('.subtotal span').innerText = subtotal;
+  return subtotal;
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+// ITERATION 2 + ITERATION 3
 
-  // ITERATION 2
-  //... your code goes here
+// code in the following two lines is added just for testing purposes.
+// it runs when only iteration 1 is completed. at later point, it can be removed.
+// const singleProduct = document.querySelector('.product');
+// updateSubtotal(singleProduct);
 
-  // ITERATION 3
-  //... your code goes here
+function calculateAll() {
+  let totalSum = 0;
+  total = document.querySelector('#total-value span');
+  const products = document.getElementsByClassName('product');
+  for (let i = 0; i < products.length; i++) {
+  //updateSubtotal(products[i]);
+  totalSum += updateSubtotal(products[i]);
```

Comment:
- A tab is missing here. The entirety of the code block should be indented.
- totalSum is a global variable. Please make it function/block scoped.